serverless/main.mk

```diff
@@ -1,26 +1,39 @@
 # Macroses
 ########################################################################################################################
+PROJ_PATH ?= $(PROJ_DIR)/$(SLS_SERVICE)-$(SLS_FUNCTION)
+
+SLS_DOCKER_IMAGE ?= amaysim/serverless
 SLS_VERSION ?= 1.73.1
-SLS_FILE ?= serverless.yml
+EVENT_FILE ?= event.json
 NODE_VERSION ?= 10.22.0-alpine3.9
 
-SLS ?= @$(DOCKER) run --entrypoint=serverless -v $(ROOT_DIR)/$(PROJECT_PATH):/opt/app -v $(HOME)/.aws/:/root/.aws \
-	-i amaysim/serverless:$(SLS_VERSION)
-NPM ?= @$(DOCKER) run --workdir=/app --entrypoint=npm -v $(ROOT_DIR)/$(PROJECT_PATH):/app -i node:$(NODE_VERSION)
+SLS ?= $(DOCKER) run --rm -e SLS_SERVICE=$(SLS_SERVICE) -e SLS_FUNCTION=$(SLS_FUNCTION) -v $(PROJ_PATH):/opt/app -v ~/.aws/:/root/.aws:ro $(SLS_DOCKER_IMAGE):$(SLS_VERSION) serverless
+NPM ?= $(DOCKER) run --rm --workdir=/app -v $(PROJ_PATH):/app node:$(NODE_VERSION) npm
+
+# Serverless CLI Reference
+########################################################################################################################
+CMD_SLS_INSTALL_PLUGINS = $(NPM) install --save-dev
+CMD_SLS_SERVICE_DEPLOY = $(SLS) deploy --verbose --stage $(ENV) --region $(AWS_REGION)
+CMD_SLS_SERVICE_INVOKE = $(SLS) invoke --function $(SLS_FUNCTION) --path $(EVENT_FILE) --stage $(ENV) --region $(AWS_REGION) --log
+CMD_SLS_SERVICE_DESTROY = $(SLS) remove --stage $(ENV) --region $(AWS_REGION)
 
 # Tasks
 ########################################################################################################################
-#aws.debug: ## Show environment information for debug purposes
-#	@echo "\033[32m=== AWS Environment Info ===\033[0m"
-#	@echo "\033[36mENV\033[0m: $(ENV)"
-#	@echo "\033[36mAWS_DEV_ENV_NAME\033[0m: $(AWS_DEV_ENV_NAME) (set devEnvironmentName here https://console.aws.amazon.com/iam/home?region=us-east-1#/users/$(AWS_USER)?section=tags)"
-#	@echo "\033[36mAWS_ACCOUNT\033[0m: $(AWS_ACCOUNT)"
-#	@echo "\033[36mAWS_PROFILE\033[0m: $(AWS_PROFILE)"
-#	@echo "\033[36mAWS_USER\033[0m: $(AWS_USER)"
-#	@echo "\033[36mTAG\033[0m: $(TAG)"
-#
-#aws.profile:
-#	$(shell mkdir -p ~/.aws && echo "[$(AWS_PROFILE)]\naws_access_key_id = $(AWS_ACCESS_KEY_ID)\naws_secret_access_key = $(AWS_SECRET_ACCESS_KEY)\nregion = $(AWS_REGION)" >> ~/.aws/credentials)
+sls.plugins:
```

Comment:
Why do we need this and not use the concept of `$(SVC)`?

Reply:
fixed